src/test/java/nl/tudelft/ewi/devhub/webtests/views/DeliveryReviewView.java

```diff
@@ -0,0 +1,196 @@
+            if (ddElements.size() == 2) {
+                ddElements.get(1).getText();
+            } else if (getDriver().findElement(By.cssSelector("blockquote dl dt")).getText().equalsIgnoreCase("Remarks")) {
+                return ddElements.get(0).getText();
+            }
+
+            return null;
+        }
+
+
+        @Nullable
+        public Date getReviewTime() throws ParseException {
+            final Optional<String> footer = getFooter();
+            if (!footer.isPresent()) {
+                return null;
+            }
+
+            return new SimpleDateFormat("EEEE dd MMMM yyyy k:m", Locale.US).parse(footer.get().split(" on ")[1].trim());
+        }
+
+        private Optional<String> getFooter() {
+            return getDriver().findElements(By.cssSelector("footer.small")).stream().findAny().map(WebElement::getText);
+        }
+    }
+
+
+    public class DeliveryForm {
+        private final WebElement anchor;
+        private final By STATE_SELECTOR = By.cssSelector("select#state");
+        private final By GRADE_SELECTOR = By.cssSelector("input#grade");
+        private final By COMMENTARY_SELECTOR = By.cssSelector("textarea#commentary");
+
+        DeliveryForm(WebElement anchor) {
+            this.anchor = anchor;
+        }
+
+        public State getSelectedState() {
+            final WebElement element = getDriver().findElement(STATE_SELECTOR);
+            Select select = new Select(element);
+
+            return State.valueOf(select.getFirstSelectedOption().getText().toUpperCase());
+        }
+
+        /**
+         * options:
+         * - Submitted
+         * - Rejected
+         * - Approved
+         * - Disapproved
+         *
+         * @param newState
+         * @throws IllegalArgumentException
+         */
+        public void setState(String newState) throws IllegalArgumentException {
+            invariant();
+            final WebElement element = getDriver().findElement(STATE_SELECTOR);
+            final Select select = new Select(element);
+
+            final Optional<WebElement> optionalNewStateElement = select.getOptions()
+                    .stream()
+                    .filter(webElement -> webElement.getText().equals(newState)).findAny();
+
+            if (optionalNewStateElement.isPresent()) {
+                optionalNewStateElement.get().click();
+            } else {
+                throw new IllegalArgumentException(newState + " is not a valid option");
+            }
+        }
+
+        public double getGrade() {
+            return Double.parseDouble(getDriver().findElement(GRADE_SELECTOR).getAttribute("value"));
+        }
+
+        public void setGrade(String newGrade) throws IllegalArgumentException {
+            invariant();
+
+            if (Double.parseDouble(newGrade) < 1D || Double.parseDouble(newGrade) > 10D) {
+                throw new IllegalArgumentException(newGrade + " is not a valid grade, grades must be between 1.0 and 10");
```

Comment:
This is actually not true. We have designed the grading system such that it supports international grading systems, and does not force you to be within `1` and `10`. Therefore we have the separation of `grade` and `state` (any of `rejected`, `approved`, `disapproved`). How the `grade` maps to the `state` (or the computed number of points through the rubrics) is course specific.